docs/en/sql-reference/functions/date-time-functions.md

```diff
@@ -601,6 +601,8 @@ toDayOfWeek(t[, mode[, timezone]])
 - `mode` - determines what the first day of the week is. Possible values are 0, 1, 2 or 3. See the table above for the differences.
 - `timezone` - optional parameter, it behaves like any other conversion function
 
+The first argument can also be specified as [String](../data-types/string.md) in a format supported by [parseDateTime64BestEffort()](type-conversion-functions.md#parsedatetime64besteffort). This was implemented for reasons of compatibility with MySQL which certain 3rd party tools expect. Because string parsing is slow, it is generally recommended to not use string arguments.
```

Comment:
I specifically wanted to emphasize _"do not use on purpose"_ and _"avoid using outside of MySQL interface"_ recommendations in the original doc entry, as we might want to consider limiting the availability of `toDayOfWeek(String)` overload to MySQL interface only as part of making MySQL interface internal implementation more tidy and less invasive.

Reply:
Okay, I'll rephrase the note to be more repulsive.